kerastuner/applications/efficientnet.py

```diff
@@ -0,0 +1,147 @@
+
+from tensorflow.keras import layers
+from tensorflow.keras.applications import efficientnet
+from tensorflow.keras.layers.experimental import preprocessing
+
+from kerastuner.engine import hypermodel
+
+
+EFFICIENTNET_MODELS = {
+    'B0': efficientnet.EfficientNetB0,
+    'B1': efficientnet.EfficientNetB1,
+    'B2': efficientnet.EfficientNetB2,
+    'B3': efficientnet.EfficientNetB3,
+    'B4': efficientnet.EfficientNetB4,
+    'B5': efficientnet.EfficientNetB5,
+    'B6': efficientnet.EfficientNetB6,
+    'B7': efficientnet.EfficientNetB7,
+}
+
+
+class HyperEfficientNet(hypermodel.HyperModel):
+    """An EfficientNet HyperModel.
+
+      # Arguments:
+
+        input_shape: shape tuple, e.g. `(256, 256, 3)`.
+              Input images will be resized if different from
+              the default input size of the version of 
+              efficientnet base model used.
+              One of `input_shape` or `input_tensor` must be
+              specified.
+        input_tensor: Keras tensor to use as image input for the model.
+              One of `input_shape` or `input_tensor` must be
+              specified.
+        classes: number of classes to classify images into.
+        augmentation: optional Model or HyperModel for image augmentation.
+        **kwargs: Additional keyword arguments that apply to all
+            HyperModels. See `kerastuner.HyperModel`.
+    """
+    def __init__(self,
+                 input_shape=None,
+                 input_tensor=None,
+                 classes=None,
+                 augmentation=None,
+                 **kwargs):
+
+        super(HyperEfficientNet, self).__init__(**kwargs)
+        if not classes:
+            raise ValueError('You must specify `classes`')
+
+        if input_shape is None and input_tensor is None:
+            raise ValueError('You must specify either `input_shape` '
+                           'or `input_tensor`.')
+
+        self.input_shape = input_shape
+        self.input_tensor = input_tensor
+        self.classes = classes
+        self.augmentation = augmentation
+
+    def build(self, hp):
+
+        if self.input_tensor is not None:
+            inputs = tf.keras.utils.get_source_inputs(self.input_tensor)
+            x = self.input_tensor
+        else:
+            inputs = layers.Input(shape=self.input_shape)
+            x = inputs
+
+        if self.augmentation:
+            if isinstance(self.augmentation, hypermodel.HyperModel):
+                augmentation = self.augmentation.build(hp)
+            elif isinstance(self.augmentation, keras.models.Model):
+                augmentation = self.augmentation
+            else:
+                raise ValueError('Keyword augment is either a HyperModel or a Keras Model')
+            x = augment(x)
+
+        # Select one of pre-trained EfficientNet as feature extractor
+        version = hp.Choice('version', ['B{}'.format(i) for i in range(8)], default='B0')
+        img_size_dict = {'B0': 224,
```

Comment:
This constant can extracted and put above.

Reply:
Sure. I moved the constant to above.